prow/jobs.yaml

```diff
@@ -176,3 +176,25 @@ kubernetes/test-infra:
   context: Jenkins go test
   rerun_command: "@k8s-bot go test this"
   trigger: "@k8s-bot (go )?test this"
+  spec:
+    nodeSelector:
+      role: build
+    containers:
+    - name: builder
+      image: gcr.io/k8s-testimages/test-infra-go-test:0.0
+      command: ["/bin/bash", "-c"]
+      args: ["git clone https://github.com/kubernetes/test-infra && ./test-infra/jenkins/bootstrap.py --repo=k8s.io/test-infra --pull=${PULL_NUMBER} --job=pull-test-infra-go-test"]
```

Comment:
Might be interesting to use a kubernetes git volume to check out test-infra, set bootstrap.py as the image command, then just provide the arguments here.